src/main/scala/io/moia/scalaHttpClient/AwsRequestSigner.scala

```diff
@@ -32,15 +33,35 @@ class AwsRequestSigner private (credentialsProvider: AwsCredentialsProvider, reg
     .putAttribute(AwsSignerExecutionAttribute.SERVICE_SIGNING_NAME, "execute-api")
     .putAttribute(AwsSignerExecutionAttribute.SIGNING_REGION, Region.of(region))
 
+  /**
+    * Signs the given HttpRequest
+    *
+    * @param request `HttpRequest` to be signed
+    * @return `Future[HttpRequest]` the signed `HttpRequest`
+    * @throws AlreadyAuthorizedException if the given `HttpRequest` already includes an "Authorize" header
+    */
   def signRequest(request: HttpRequest): Future[HttpRequest] =
-    Future {
-      val sdkRequest = toSdkRequest(request)
-      val signedSdkRequest = blocking {
-        awsSigner.sign(sdkRequest, executionAttributes)
-      }
-      // construct new HttpRequest with signed URI query params and headers
-      HttpRequest(request.method, uri = Uri(signedSdkRequest.getUri.toString), getSdkHeaders(signedSdkRequest), request.entity)
-    }(mat.executionContext)
+    if (isAlreadyAuthorized(request.headers))
+      Future.failed(AlreadyAuthorizedException())
+    else
+      Future {
+        val sdkRequest = toSdkRequest(request)
+        val signedSdkRequest = blocking {
+          awsSigner.sign(sdkRequest, executionAttributes)
+        }
+        // construct new HttpRequest with signed URI query params and headers
+        HttpRequest(request.method, uri = Uri(signedSdkRequest.getUri.toString), getSdkHeaders(signedSdkRequest), request.entity)
+      }(mat.executionContext)
+
+  /**
+    * Checks if the given collection of `HttpHeader`s includes one "Authorization" header
+    * @param headers `Seq[HttpHeader]` headers of an Akka `HttpRequest`
+    * @return true if "Authorization" header exists
+    */
+  private def isAlreadyAuthorized(headers: Seq[HttpHeader]): Boolean =
+    headers.exists { header: HttpHeader =>
+      header.is("authorization")
```

Comment:
`HttpHeader` is always case-insensitive. In this case, the `is` method requires you to supply the `lowerCaseName`.

Reply:
I don't like the `req => sendRequest(req)` thing because `sendRequest` would work as well but Scapegoat complains. I want to understand why it doesn't complain for `strictify` for example.